### PR TITLE
Add user friendly message for failed Deepl API requests

### PR DESCRIPTION
**Pull request by MizukiTemma**
### Short description

This PR adds user friendly error messages for DeepL errors.

### Proposed changes

- add `try` and `except` with new messages

### Side effects

- I'll wait for #2606 to make it sure that new error messages won't spam users.

### Resolved issues

Fixes: #2280 

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by MizukiTemma** [comment]
Currently there is a detaield message for each case listed [here](https://github.com/digitalfabrik/integreat-cms/issues/2280#issuecomment-1551639085)


Though I and @david-venhoff came to the idea hat the error message probably doesn't have to be so detailed as it is right now and one universal error message is enough, for the focus of the problem is that the users get interrupted by the untreated exceptions.

What do you think @osmers ?

**Comment by timobrembeck** [comment]
@MizukiTemma @osmers yes I agree, we don't necessarily need the detailed errors, but it helps when users report the problem to us, so we can easier identify and reproduce the problem ourselves.

File: integreat_cms/locale/de/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -9609,6 +9609,12 @@ msgstr[1] ""
 "Die folgenden {model_name_plural} konnten nicht automatisch übersetzt "
 "werden: {object_names}"
 
+#: deepl_api/deepl_api_client.py
+msgid ""
+"An problem with DeepL API has occurred. Please contact an administrator."
```
Comment (david-venhoff) [inline integreat_cms/locale/de/LC_MESSAGES/django.po:9614]:
```suggestion
"A problem with DeepL API has occurred. Please contact an administrator."
```

File: tests/deepl_api/deepl_api_test.py [hunk 2]
```diff
@@ -1048,14 +1048,20 @@ def test_deepl_bulk_mt_no_target_language(
...
 @pytest.mark.parametrize(
     "login_role_user", PRIV_STAFF_ROLES + [AUTHOR, MANAGEMENT, EDITOR], indirect=True
 )
-@pytest.mark.skip(reason="server error handling is currently not implemented")
+@pytest.mark.parametrize("api_errors", api_errors)
+# pylint: disable=redefined-outer-name
 def test_deepl_bulk_mt_api_error(
     load_test_data: None,
     login_role_user: tuple[Client, str],
+    api_errors: int,
```
Comment (david-venhoff) [inline tests/deepl_api/deepl_api_test.py:1064]:
```suggestion
    error: int,
```
A bit nitpicky, but maybe you could rename this to `error`, since right now it sounds like it could store multiple errors

**Review by david-venhoff — APPROVED** [review]
Thanks, LGTM

File: integreat_cms/deepl_api/deepl_api_client.py [hunk 3]
```diff
@@ -147,13 +150,18 @@ def translate_queryset(
...
+                            data[attr] = self.translator.translate_text(
+                                unescape(getattr(source_translation, attr)),
+                                source_lang=source_language.slug,
+                                target_lang=target_language_key,
+                                tag_handling="html",
+                            )
+                        except DeepLException as e:
+                            deepl_api_error = True
+                            logger.error(e)
+                            break
```
Comment (seluianova) [inline integreat_cms/deepl_api/deepl_api_client.py:164]:
```suggestion
                        except DeepLException as e:
                            messages.error(
                                self.request,
                                _(
                                    "A problem with DeepL API has occurred. Please contact an administrator."
                                ),
                            )
                            logger.error(e)
                            return
```

Assuming that we don't need 2 error messages here?
image
The top message occurs because the condition `if content_translation_form.is_valid()` is not met. It also adds the following message to the log:
> Mar 26 10:01:35 ERROR integreat_cms.deepl_api.deepl_api_client - Automatic translation for <Page (id: 4, parent: 1, region: augsburg, slug: interesting-facts-about-augsburg)> could not be created because of {'title': ['This field is required.'], 'slug': ['Cannot generate slug from title.']}
Seems irrelevant to me when there's a deepl error.

**Comment by MizukiTemma** [comment]
@seluianova 
Thank you 😸 I've adopted your idea

**Review by seluianova — APPROVED** [review]
LGTM, thanks!
